File: tic_tac_node/src/player.js
export default class Player {
    constructor(name, mark) {
        this.name = name;
        this.mark = mark;
    }

    getName() {
        return this.name;
    }

    getMark() {
        return this.mark;
    }

    makeMove(board, row, col) {
        try {
            board.setCell(row, col, this.mark);
            return true;
        } catch (_) {
            return false;
        }
    }
};

File: tic_tac_rs/src/player.rs
use crate::utils::Mark;
use crate::board::{Board, BoardTrait};

pub trait PlayerTrait {
    fn get_name(&self) -> &str;
    fn get_mark(&self) -> &Mark;
    fn make_move(&self, board: &mut Board, row: usize, col: usize) -> bool;
}

#[derive(Clone, PartialEq, Eq)]
pub struct Player {
    pub name: String,
    pub mark: Mark,
}

impl PlayerTrait for Player {
    fn get_name(&self) -> &str {
        &self.name
    }

    fn get_mark(&self) -> &Mark {
        &self.mark
    }

    fn make_move(&self, board: &mut Board, row: usize, col: usize) -> bool {
        board.set_cell(row, col, self.mark);
        true
    }
}